server/src/main/java/io/unitycatalog/server/service/TemporaryTableCredentialsService.java

```diff
@@ -29,8 +33,15 @@ public TemporaryTableCredentialsService(CredentialOperations credentialOps) {
   }
 
   @Post("")
+  @AuthorizeExpression("""
+          #authorize(#principal, #metastore, OWNER) ||
+          #authorize(#principal, #catalog, OWNER) ||
+          (#authorize(#principal, #schema, OWNER) && #authorize(#principal, #catalog, USE_CATALOG)) ||
+          (#authorize(#principal, #schema, USE_SCHEMA) && #authorize(#principal, #catalog, USE_CATALOG) && #authorizeAny(#principal, #table, OWNER, SELECT))
```

Comment:
Similar to model temp credentials, the required privileges for table credentials depend on the operations -- select for read privileges and select+modify for write privileges. The rules are:

For read operation:
```
#authorizeAny(#principal, #schema, OWNER, USE_SCHEMA) && #authorize(#principal, #catalog, OWNER, USE_CATALOG) && #authorizeAny(#principal, #table, OWNER, SELECT)
```

For read/write:
```
#authorizeAny(#principal, #schema, OWNER, USE_SCHEMA) && 
#authorize(#principal, #catalog, OWNER, USE_CATALOG) && 
(#authorize(#principal, #table, OWNER) || #authorizeAll(#principal, #table, SELECT, MODIFY))
```